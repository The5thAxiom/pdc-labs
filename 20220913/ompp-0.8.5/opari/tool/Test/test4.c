#include <stdio.h>

int main() {
  int i;

  #pragma omp parallel
  {
    printf("parallel\n");
    #pragma omp for nowait
    for (i=0; i<4; ++i) printf("do ; ; ; nowait %d\n", i);

    #pragma omp barrier

    #pragma omp \
	   /* unnütz */ \
	   /* mehr unnütz */ \
	   for /* KEIN nowait */
    for (i=0; i<4; ++i) {
      printf("do nowait? %d\n", i);
    }

    #pragma omp for
    for (i=0; i<4; ++i) {
      printf("do %d\n", i);
    }
  }
}
