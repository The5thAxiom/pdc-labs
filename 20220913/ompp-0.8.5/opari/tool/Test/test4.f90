       program test4
       integer i

       !$omp parallel
       write(*,*) "parallel"
         !$omp do
         do i=1,4
           write(*,*) "do nowait",i
         enddo
         !$omp enddo nowait

         !$omp barrier

         !$omp do
         do i=1,4
           write(*,*) "do nowait?",i
         enddo
         !$omp   end   &  ! continue
         !$omp &       &  ! still continue
         !$omp & do       ! kein nowait

         !$omp do
         do i=1,4
           write(*,*) "do",i
         enddo
         !$omp end do
       !$omp end parallel

       end
