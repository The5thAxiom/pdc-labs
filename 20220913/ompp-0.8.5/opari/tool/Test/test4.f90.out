#ifdef _POMP
#  undef _POMP
#endif
#define _POMP 200110

#line 1 "test4.f90"
       program test4
       integer i

      call POMP_Parallel_fork(18)
#line 4 "test4.f90"
       !$omp parallel
      call POMP_Parallel_begin(18)
#line 5 "test4.f90"
       write(*,*) "parallel"
      call POMP_Do_enter(19)
#line 6 "test4.f90"
         !$omp do
         do i=1,4
           write(*,*) "do nowait",i
         enddo
#line 10 "test4.f90"
         !$omp enddo nowait
      call POMP_Do_exit(19)
#line 11 "test4.f90"

      call POMP_Barrier_enter(20)
#line 12 "test4.f90"
         !$omp barrier
      call POMP_Barrier_exit(20)
#line 13 "test4.f90"

      call POMP_Do_enter(21)
#line 14 "test4.f90"
         !$omp do
         do i=1,4
           write(*,*) "do nowait?",i
         enddo
#line 18 "test4.f90"
         !$omp   end   &  ! continue
         !$omp &       &  ! still continue
         !$omp & do        nowait! kein nowait
      call POMP_Barrier_enter(21)
!$omp barrier
      call POMP_Barrier_exit(21)
      call POMP_Do_exit(21)
#line 21 "test4.f90"

      call POMP_Do_enter(22)
#line 22 "test4.f90"
         !$omp do
         do i=1,4
           write(*,*) "do",i
         enddo
#line 26 "test4.f90"
         !$omp end do nowait
      call POMP_Barrier_enter(22)
!$omp barrier
      call POMP_Barrier_exit(22)
      call POMP_Do_exit(22)
#line 27 "test4.f90"
      call POMP_Barrier_enter(18)
!$omp barrier
      call POMP_Barrier_exit(18)
      call POMP_Parallel_end(18)
#line 27 "test4.f90"
       !$omp end parallel
      call POMP_Parallel_join(18)
#line 28 "test4.f90"

       end
